src/arithmetic/arithmetic_expression.c

```diff
@@ -592,6 +558,28 @@ bool inline AR_EXP_IsParameter(const AR_ExpNode *exp) {
 	return exp->type == AR_EXP_OPERAND && exp->operand.type == AR_EXP_PARAM;
 }
 
+bool AR_EXP_IsAttribute(const AR_ExpNode *exp, char **attr) {
+	ASSERT(exp != NULL);
+
+	// an arithmetic expression performs attribute extraction
+	// if it applys the "property" function, in which case the left-handside
+	// child represents the graph entity from which we access the attribute
+	// while the right-handside represents the attribute name
+
+	if(exp->type != AR_EXP_OP) return false;
+	if(RG_STRCMP(exp->op.func_name, "property") != 0) return false;
```

Comment:
I would write these as a single `||`-separated if condition, but it's probably just an aesthetic choice.